### PR TITLE
CEDS-2688 Upgrade Scala version to 2.12.12

### PR DESCRIPTION
**Pull request by MaciejRewera**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -75,4 +77,15 @@ lazy val scoverageSettings: Seq[Setting[_]] = Seq(
   coverageFailOnMinimum := true,
   coverageHighlighting := true,
   parallelExecution in Test := false
-)
\ No newline at end of file
+)
+
+lazy val silencerSettings: Seq[Setting[_]] = {
```
Comment (SteveSugden) [inline build.sbt:82]: Are these the settings we need @PatrykRudnicki to get rid of our warnings?

**Review by SteveSugden — APPROVED** [review]
I'm no export, but LGTM

File: build.sbt (hunk 1 above)
Comment (MaciejRewera) [reply build.sbt:82]: I've used them to silence dozens of warning coming from twirl templates and `.routes` files.

Comment (PatrykRudnicki) [reply build.sbt:82]: Yes, it will reduce amount of warnings

File: app/models/cache/Cache.scala [hunk 2]
```diff
@@ -27,20 +27,6 @@ case class Cache(providerId: String, answers: Option[Answers], queryUcr: Option[
 }
 
 object Cache {
-  implicit private val formatInstant: OFormat[Instant] = new OFormat[Instant] {
```
Comment (PatrykRudnicki) [inline app/models/cache/Cache.scala:30]: Are you sure this is not required?

I remember that we had a problem with Instant, but not sure what was the problem.

**Review by PatrykRudnicki — APPROVED** [review]
Only worried about Instant format, but if everything works like excepted and ATs passed that should be okay

File: app/models/cache/Cache.scala (hunk 2 above)
Comment (MaciejRewera) [reply app/models/cache/Cache.scala:30]: All tests passed (ATs too). Might not be an issue anymore.
Removed that after being pointed out by the compiler as unused.

**Pull request merged**